### ISSUE TITLE
Has anyone tested this net? Does it actually work?

### ISSUE THREAD

**Issue by Rizhiy** [PureDiors/pytorch_RFCN#4]
I have tried to train it, but got just noise as the output. Has anyone managed to train it?

**Comment by strongwolf** [comment]
What is your mAP? I also have tried to train it but result in a very low accuracy.

**Comment by Rizhiy** [comment]
mAP is almost zero. I have tried to debug and it appears that bbox regressor doesn't work at all.
I think that PSRoIPooling might not be implemented correctly here.

**Comment by noUmbrella** [comment]
Hi, I find that this code has a serious bug in rfcn.py at 212 and 213, I think it should be self.rfcn_bbox = Conv2d(1024,7*7*8, 1,1, bn=False)   self.rfcn_score = Conv2d(1024, 7*7*self.n_classes,1,1,bn=False), do you change here? If you do that, can it work well?

**Comment by noUmbrella** [comment]
Hi, I find that this code has a serious bug in rfcn.py at 212 and 213, I think it should be self.rfcn_bbox = Conv2d(1024,778, 1,1, bn=False) self.rfcn_score = Conv2d(1024, 77self.n_classes,1,1,bn=False), do you change here? If you do that, can it work well?@Rizhiy

**Comment by RichardMrLu** [comment]
@noUmbrella  after your solved this bug, do you work well? what is your mAP?

**Comment by noUmbrella** [comment]
I test the mAP is very small, about 0.25. I found that, the classify is right and the regression is not work@RichardMrLu

**Comment by Rizhiy** [comment]
I have the same experience as @noUmbrella, classification works fine, but bbox regression is very bad.

**Comment by noUmbrella** [comment]
Hi, do you find the bug? Does the Position Pooling Layer exists problem? @Rizhiy

**Comment by noUmbrella** [comment]
Hi, do you find the bug? Does the Position Pooling Layer exists problem? @RichardMrLu

**Comment by RichardMrLu** [comment]
@noUmbrella  = = sorry, I didn't run the code.

**Comment by jjprincess** [comment]
I have changed self.rfcn_bbox and  self.rfcn_score ,but when i trained the network,bbox regression can not converge.Does anybody can help?

**Comment by Rizhiy** [comment]
@jjprincess For me, it also either doesn't converge or converges poorly and I get very low mAP.

**Comment by jjprincess** [comment]
@Rizhiy There may  be some wrong with the code,I has changed the frame “mxnet”,I got right results.
Here is the link"https://github.com/msracver/Deformable-ConvNets"

**Comment by Rizhiy** [comment]
@jjprincess Yes, I'm already using that version, but I like PyTorch more than MXNet.

**Comment by sophieyl** [comment]
I use VOC2007 so I change the imdb_name='voc_2007_trainval' in train.py in line36.
when i debug, I find that in pascal_voc.py line 49, self._classes had been rewrite, so I comment that line and then change line 212 	self.rfcn_score = Conv2d(1024,7*7*21, 1,1, bn=False)      in file rfcn.py @noUmbrella

**Comment by lxtGH** [comment]
look at this [repo](https://github.com/xingmimfl/pytorch_RFCN)

**Comment by princevandev** [comment]
This repo stop updating for more than one year. If someone still interesting in pytorch R-FCN, pls look at another pytorch implement - [princewang1994/R-FCN.pytorch](https://github.com/princewang1994/R-FCN.pytorch), thanks~
